### PR TITLE
[PR] Make some empty class and struct for interpretor

### PR DESCRIPTION
**Pull request by tucan9389**
## PR Point

- Add `TFLiteImageInterpretor` for implementing pre-process of image input
- Add `PoseEstimator` protocol for abstraction
- Add `PoseNetPoseEstimator` for specifying input/output and model name of PoseNet tflite file

### CODE REVIEW

File: PoseEstimation-TFLiteSwift/PoseEstimator.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    init(tfliteResult: TFLiteResult) {
+        let heatmaps = Heatmaps(tfliteResult: tfliteResult)
+        let keypoints = heatmaps.keypointsAndScore.map { Keypoint(position: $0.position, score: $0.score) }
+        // <#TODO#>
+        self = .success(keypoints: keypoints, heatmaps: heatmaps)
+    }
+}
+
+protocol PoseEstimator {
+    func predict(with pixelBuffer: CVPixelBuffer) -> PoseEstimationResult
```
Comment (syjdev) [inline PoseEstimation-TFLiteSwift/PoseEstimator.swift]: predict라는 것은 '예측'인데, 이 함수를 구현한 쪽에서는... 하려는 동작이 '예측'과는 다소 다른 것 같아요.

어짜피 리턴해주는 것도 estimation에 대한 결과이니 estimate가 어떨까요?
아니면 inference, 같은것도 괜찮을 것 같고

File: PoseEstimation-TFLiteSwift/ViewController.swift [hunk 2]
```diff
@@ -9,12 +9,11 @@
 import UIKit
 
 class ViewController: UIViewController {
+    
+    let poseEstimator: PoseEstimator = PoseNetPoseEstimator()
```
Comment (syjdev) [inline PoseEstimation-TFLiteSwift/ViewController.swift]: What is the mean of NetPose?

File: PoseEstimation-TFLiteSwift/PoseEstimator.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+//  PoseEstimator.swift
+//  PoseEstimation-TFLiteSwift
+//
+//  Created by Doyoung Gwak on 2020/03/14.
+//  Copyright © 2020 Doyoung Gwak. All rights reserved.
+//
+
+import CoreVideo
+
+enum PoseEstimationResult {
```
Comment (syjdev) [inline PoseEstimation-TFLiteSwift/PoseEstimator.swift]: have you ever seen, https://developer.apple.com/documentation/swift/result ?
I think, you need to implement no fail case.

File: PoseEstimation-TFLiteSwift/ViewController.swift (hunk 2 above)
Comment (tucan9389) [reply PoseEstimation-TFLiteSwift/ViewController.swift]: [`PoseNet`](https://arxiv.org/abs/1505.07427) is the paper for pose estimation distributed in 2015. `PoseNetPoseEstimator` stand for `PoseNet` + `PoseEstimator` which means that the pose estimator implementing `PoseNet` tflite model.

File: PoseEstimation-TFLiteSwift/PoseEstimator.swift (hunk 3 above)
Comment (tucan9389) [reply PoseEstimation-TFLiteSwift/PoseEstimator.swift]: Cool. Thanks for your advice.

File: PoseEstimation-TFLiteSwift/PoseEstimator.swift [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        let score: CGFloat
+    }
+    struct Heatmaps {
+        // <#TODO#>
+        
+        init(tfliteResult: TFLiteResult) {
+            // <#TODO#>
+        }
+        
+        var keypointsAndScore: [(position: CGPoint, score: CGFloat)] {
```
Comment (syjdev) [inline PoseEstimation-TFLiteSwift/PoseEstimator.swift]: Keypoint have a score.
So, keypointsAndScore have duplicated meaning.

And I can implement, var keypoints: [Keypoint] { ... }.
This is more simple, isn't it?

File: PoseEstimation-TFLiteSwift/ViewController.swift (hunk 2 above)
Comment (syjdev) [reply PoseEstimation-TFLiteSwift/ViewController.swift]: Thank, I undetstood.

File: PoseEstimation-TFLiteSwift/PoseEstimator.swift (hunk 1 above)
Comment (tucan9389) [reply PoseEstimation-TFLiteSwift/PoseEstimator.swift]: Thanks for your suggestion. I'll change to `inference`.

File: PoseEstimation-TFLiteSwift/PoseEstimator.swift (hunk 4 above)
Comment (tucan9389) [reply PoseEstimation-TFLiteSwift/PoseEstimator.swift]: @syjdev 
👍 Right, I replaced to `keypoints: [Keypoints]` property.

**Comment by tucan9389** [comment]
@syjdev 
Thanks for your reviews

**Pull request merged**